[material-ios] Updated the binding to v72.2.0

mattleibow opened the pull request:
Updated to https://github.com/material-components/material-components-ios/releases/tag/v72.1.0

> VS bug [#747237](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/747237)

mattleibow committed:
[material-ios] Updated the binding to v70.1.0

mattleibow committed:
[material-ios] Updated the binding to v72.1.0

PureWeen commented:
Version 72 Installed and ran fine with Visual Gallery locally
Only thing I had to change was the call to ```SetShadowElevation``` which now takes an nfloat opposed to a double

mattleibow commented:
Yeah, they changed the type of all things elevated to `CGFloat` or our `nfloat`.

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:
```diff
@@ -453,33 +661,126 @@ interface AppBar {
-	// @interface MDCAppBarTextColorAccessibilityMutator : NSObject
-	[Obsolete ("This API will be deprecated with no replacement.")]
-	[BaseType (typeof (NSObject), Name = "MDCAppBarTextColorAccessibilityMutator")]
-	interface AppBarTextColorAccessibilityMutator {
```
The mutator is dead!

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:
```diff
@@ -453,33 +661,126 @@ interface AppBar {
 		// @property (assign, nonatomic) MDCShadowElevation elevation;
 		[Export ("elevation")]
-		double Elevation { get; set; }
+		nfloat Elevation { get; set; }
```
The `MDCShadowElevation` has been mutated into a `CGFloat`

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:
```diff
@@ -945,8 +1434,15 @@ interface Button : INativeObject {
+	[BaseType (typeof (UIView),
+		Name = "MDCButtonBar",
+		Delegates = new [] { "Delegate" },
+		Events = new [] { typeof (ButtonBarDelegate) })]
```
They added a `delegate` property, so I added our magic.

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:
```diff
@@ -4777,110 +5411,105 @@ interface RoundedCornerTreatment {
-	interface SemanticColorScheme : ColorScheming {
-		// @property (readwrite, nonatomic) UIColor * _Nonnull primaryColor;
-		[Export ("primaryColor", ArgumentSemantic.Assign)]
+	interface SemanticColorScheme : ColorScheming, INSCopying {
```
Some schemes are now copy-stuff.

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:
```diff
@@ -4893,101 +5522,6 @@ interface SemanticColorScheme : ColorScheming {
 	}
 
-	[Static]
-	interface ShadowElevations {
```
This whole type has been moved to a const enum, and no longer fields

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:
```diff
@@ -5080,22 +5618,13 @@ interface ShapedView {
-
-		// -(instancetype _Nonnull)initWithFrame:(CGRect)frame;
-		[Export ("initWithFrame:")]
-		IntPtr Constructor (CGRect frame);
```
This was removed, and I added it back in the extensions file

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/Extensions.cs:
```diff
@@ -2,11 +2,89 @@
 
 namespace MaterialComponents {
+
+	public static class ShadowElevations {
```
The new consts

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/Extensions.cs:
```diff
@@ -35,20 +113,22 @@ public static class MaterialFeatureHighlightStrings {
-					semanticColorScheme = Runtime.GetNSObject<SemanticColorScheme> (ColorScheme.Handle, false);
-
-				return semanticColorScheme;
+				return Runtime.GetNSObject<SemanticColorScheme> (ColorScheme.Handle, false);
```
I may be wrong, but we shouldn't cache this as changing the other property will break this? This is supposed to be the same object - just a more type-specific version. ie: SemanticColorScheme === ColorScheme

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/Extensions.cs:
```diff
@@ -35,20 +113,22 @@ public static class MaterialFeatureHighlightStrings {
 				ColorScheme = value;
 			}
 		}
+		//public ShapeScheme ShapeScheme {
```
I couldn't think of another name for this - the types have changed, but are just different with a new `I`

mattleibow commented on iOS/MaterialComponents/source/MaterialComponents/Properties/AssemblyInfo.cs:
```diff
@@ -17,15 +17,16 @@
 [assembly: AssemblyCompany ("")]
 [assembly: AssemblyProduct ("")]
-[assembly: AssemblyCopyright ("(c) Bill Holmes")]
+[assembly: AssemblyCopyright ("© Microsoft Corporation. All rights reserved.")]
```
@bholmes 🥇

mattleibow committed:
[material-ios] Fixed a few things
 - the method names were not nice
 - the overloads that were `Class<XXX>` were mapped to `XXX`
    - fixed to take `Class`
    - added overloads that take a .NET `Type` and a generic

mattleibow committed:
[material-ios] Fixed a copy-paste error

mattleibow committed:
[material-ios] Updated the binding to v72.2.0

mattleibow commented:
LOL, another update :)
